sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -282,4 +284,45 @@ namespace Azure { namespace Core { namespace IO {
     int64_t Length() const override;
   };
 
+  /**
+   * @brief A concrete implementation of #Azure::Core::IO::BodyStream that wraps another stream and
+   * reports progress
+   */
+  class ProgressBodyStream : public BodyStream {
+  private:
+    BodyStream* m_bodyStream;
+    std::atomic_int64_t m_position;
```

Comment:
The whole body stream thing is not thread-safe, which means users are not supposed to call `Read` from more than one thread at a time. So what's the point of using atomic variable here?

Reply:
you could have the the reads on one thread and the callback on another, this way reads and writes should be ok

Reply:
No, this shouldn't really happen and an int64_t can easily be copied across thread so this field should not be atomic.

Reply:
done, make into  int_64